pkg/bytecode/compiler.go

```diff
@@ -7,9 +7,14 @@ import (
 	"evylang.dev/evy/pkg/parser"
 )
 
-// ErrUndefinedVar is returned when a variable name cannot
-// be resolved in the symbol table.
-var ErrUndefinedVar = errors.New("undefined variable")
+var (
+	// ErrUndefinedVar is returned when a variable name cannot
+	// be resolved in the symbol table.
+	ErrUndefinedVar = errors.New("undefined variable")
```

Comment:
`ErrUndefinedVar` should wrap a `ErrRuntime` or `ErrPanic` or similar:
```go
ErrUndefinedVar = fmt.Errorf("%w: undefined variable", ErrRuntime)
```

This specific error can actually happen in parsed evy code (it is not an internal error), e.g.:

```
b := makeNum
print "b" b
a := 3

func makeNum:num
    return a
end
```

A better parser (like Go's) should find this at parse time, but ours does not at the moment.